proxy/src/http/source.rs

```diff
@@ -344,26 +344,40 @@ mod handler {
         project_urn: String,
     ) -> Result<impl Reply, Rejection> {
         let peer = peer.lock().await;
-        let (branches, tags) = peer
-            .with_browser(&project_urn, |browser| {
-                Ok((coco::branches(browser)?, coco::tags(browser)?))
-            })
-            .await?;
 
-        let revs = ["cloudhead", "rudolfs", "xla"]
-            .iter()
-            .map(|handle| super::Revision {
-                branches: branches.clone(),
-                tags: tags.clone(),
-                identity: identity::Identity {
-                    id: format!("{}@123abcd.git", handle),
-                    metadata: identity::Metadata {
-                        handle: (*handle).to_string(),
+        let project = peer.get_project(&project_urn).await?;
+        let api = peer.api.lock().unwrap();
+        let peer_id = api.peer_id();
+        let storage = api.storage();
+        let repo = storage.open_repo(project.urn()).unwrap();
+        let refs = repo.rad_refs().unwrap();
+
+        let remotes = refs.remotes.flatten().map(|remote| {
+            let refs = if remote == &peer_id {
+                let browser = repo.browser(project.default_branch()).unwrap();
+                coco::local_branches(&browser).unwrap()
+            } else {
+                let refs = storage.rad_refs_of(&project.urn(), remote.clone()).unwrap();
+                refs.heads.keys().cloned().map(coco::Branch).collect()
+            };
+
+            (remote, refs)
+        });
```

Comment:
@kim when implementing this, we ran into the fact that `rad_refs().remotes` includes your own PeerId in the list of remotes. We work around this by checking whether it's us, and if so, getting local branches from surf. Is there a better way to use this API?